sokutei.h

```diff
@@ -0,0 +1,194 @@
+#ifndef SOKUTEI_BENCHMARK_H
+#define SOKUTEI_BENCHMARK_H
+
+#define SOKUTEI_MAX_MEASURED_ITERATIONS 10
+#define SOKUTEI_MAX_COUNTER_COUNT  100
+#define SOKUTEI_MAX_COUNTER_NAME_LENGTH 30
+#define SOKUTEI_INTEGER_COUNTER_TYPE int
+#define SOKUTEI_FLOAT_COUNTER_TYPE double
+#define SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE int
+
+#define MAX_SIZE_OF_TYPES   ((sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_FLOAT_COUNTER_TYPE)) \
+                            ? ((sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)) \
+                                ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE))\
+                            : ((sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)) \
+                                ? sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)))
+
+#define SOKUTEI_INTEGER_TYPE '1'
+#define SOKUTEI_FLOAT_TYPE '2'
+#define SOKUTEI_INTERVAL_TYPE '3'
+
+#define SOKUTEI_NOT_FOUND -1
+#define SOKUTEI_NOT_MATCHING_TYPE -2
+#define SOKUTEI_COUNTER_LIMIT_REACHED -3
+
+
+#ifdef SOKUTEI_LOGGING_ENABLED
+#define SOKUTEI_DEBUG_PRINT(x, y) printf(x, y)
+#else
+#define SOKUTEI_DEBUG_PRINT(x, y)
+#endif
+
+
+int sokutei_strcmp(const char *string_a, const char *string_b){
+    int i = 0;
+    while(string_a[i] == string_b[i] && string_a[i]) i++;
+    return string_a[i] - string_b[i];
+}
+
+
+char *sokutei_strcpy(char *string_a, const char *string_b){
+    int i;
+    for(i = 0; (string_a[i] = string_b[i]) && (i < SOKUTEI_MAX_COUNTER_NAME_LENGTH); i++);
+    string_a[SOKUTEI_MAX_COUNTER_NAME_LENGTH] = '\0';
+    return string_a;
+}
+#define sokutei_string_equals(string_a, string_b) (sokutei_strcmp(string_a, string_b) == 0)
+
+
+int sokutei_number_of_iterations = SOKUTEI_MAX_MEASURED_ITERATIONS;
+
+/**
+ * sokutei_counter_definitions is an array consist of the counter name followed by string separator (\0), and the type of the counter
+ */
+char sokutei_counter_definitions[SOKUTEI_MAX_COUNTER_COUNT][SOKUTEI_MAX_COUNTER_NAME_LENGTH + 2] = {'\0'};
+
+
+char sokutei_counters[SOKUTEI_MAX_COUNTER_COUNT * MAX_SIZE_OF_TYPES * SOKUTEI_MAX_MEASURED_ITERATIONS] = {0};
+int  sokutei_number_of_counters = 0;
+int number_of_iterations = SOKUTEI_MAX_MEASURED_ITERATIONS;
+
+
+char sokutei_get_type_of(const int index){
+    return sokutei_counter_definitions[index][31];
```

Comment:
This `31` stuff looks quite like a magic number, isn't it? What do you think, how can we address this?